.specify/scripts/bash/update-agent-context.sh

```
@@ -0,0 +1,756 @@
+        elif [[ $in_changes_section == true ]] && [[ "$line" =~ ^##[[:space:]] ]]; then
+            # Before closing the section, add new change if not already present
+            if [[ $changes_entries_added == false ]] && [[ -n "$new_change_entry" ]] && [[ $skip_adding_new_change == false ]]; then
+                echo "$new_change_entry" >> "$temp_file"
+                changes_entries_added=true
+            fi
+            echo "$line" >> "$temp_file"
+            in_changes_section=false
+            continue
+        elif [[ $in_changes_section == true ]] && [[ "$line" == "- "* ]]; then
+            # Trim whitespace/newlines for comparison
+            local trimmed_line=$(echo "$line" | sed 's/^[[:space:]]*//;s/[[:space:]]*$//')
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)